plugins/base/src/main/kotlin/translators/descriptors/DescriptorAccessorConventionUtil.kt

```diff
@@ -0,0 +1,79 @@
+package org.jetbrains.dokka.base.translators.descriptors
+
+import org.jetbrains.dokka.base.translators.firstNotNullOfOrNull
+import org.jetbrains.kotlin.descriptors.FunctionDescriptor
+import org.jetbrains.kotlin.descriptors.PropertyDescriptor
+import org.jetbrains.kotlin.load.java.JvmAbi
+import org.jetbrains.kotlin.load.java.descriptors.JavaMethodDescriptor
+import org.jetbrains.kotlin.load.java.propertyNameByGetMethodName
+import org.jetbrains.kotlin.load.java.propertyNamesBySetMethodName
+
+internal data class DescriptorFunctionsHolder(
+    val regularFunctions: List<FunctionDescriptor>,
+    val accessors: Map<PropertyDescriptor, List<FunctionDescriptor>>
+)
+
+internal fun splitFunctionsAndAccessors(
+    properties: List<PropertyDescriptor>,
+    functions: List<FunctionDescriptor>
+): DescriptorFunctionsHolder {
+    val fieldsByName = properties.associateBy { it.name.asString() }
+    val regularFunctions = mutableListOf<FunctionDescriptor>()
+    val accessors = mutableMapOf<PropertyDescriptor, MutableList<FunctionDescriptor>>()
+    functions.forEach { function ->
+        val possiblePropertyNamesForFunction = function.toPossiblePropertyNames()
+        val field = possiblePropertyNamesForFunction.firstNotNullOfOrNull { fieldsByName[it] }
+        if (field != null) {
+            accessors.getOrPut(field, ::mutableListOf).add(function)
+        } else {
+            regularFunctions.add(function)
+        }
+    }
+    return DescriptorFunctionsHolder(regularFunctions, accessors)
+}
+
+internal fun FunctionDescriptor.toPossiblePropertyNames(): List<String> {
+    val stringName = this.name.asString()
+    return when {
+        JvmAbi.isSetterName(stringName) -> propertyNamesBySetMethodName(this.name).map { it.asString() }
+        JvmAbi.isGetterName(stringName) -> propertyNamesByGetMethod(this)
+        else -> listOf()
+    }
+}
+
+internal fun propertyNamesByGetMethod(functionDescriptor: FunctionDescriptor): List<String> {
+    val stringName = functionDescriptor.name.asString()
+    // In java, the convention for boolean property accessors is as follows:
+    // - `private boolean active;`
+    // - `private boolean isActive();`
+    //
+    // Whereas in Kotlin, because there are no explicit accessors, the convention is
+    // - `val isActive: Boolean`
+    //
+    // This makes it difficult to guess the name of the accessor property in case of Java
+    val javaPropName = if (functionDescriptor is JavaMethodDescriptor && JvmAbi.startsWithIsPrefix(stringName)) {
+        val javaPropName = stringName.removePrefix("is").let { newName ->
+            newName.replaceFirst(newName[0], newName[0].toLowerCase())
+        }
+        javaPropName
+    } else {
+        null
+    }
+    val kotlinPropName = propertyNameByGetMethodName(functionDescriptor.name)?.asString()
```

Comment:
In the example you gave - sure, but not in this particular place and it's not an accessor by current definition for multiple reasons:
1. `val v` is immutable, should be `var`
2. `setV` is not a proper setter since it doesn't have an argument. Should be `fun setV(v: Int)`, otherwise this could be just a simple method
3. If you change change the example to account for 1 and 2, the code will not compile with declaration clash error. I believe you can't write an explicit accessor in Kotlin, `get()` and `set()` should be used, but I might be wrong
4. Last but not least, `val v` is public, there's a check down below that that accessors should not be public api.

Reply:
>  it's not an accessor

Yes, it was made intentionally. 
In my example the **regular** function  `setV()` will be lost in documentation.

Reply:
Wow, nice! I totally missed that. Added a fix + tests